keps/sig-cluster-lifecycle/kubeadm/2568-kubeadm-non-root-control-plane/README.md

```diff
@@ -0,0 +1,1069 @@
+      runAsGroup: 2000  # this value is only an example and is not the id we plan to use.
+      allowPrivilegeEscalation: false
+      seccompProfile:
+        type: runtime/default
+      capabilities:
+        drop:
+        - ALL
+    image: k8s.gcr.io/kube-apiserver:v1.21.0-beta.0.368_9850bf06b571d5-dirty
+    ... # omitted to save space
+    volumeMounts:
+    ... # omitted to save space
+    - mountPath: /etc/kubernetes/pki
+      name: k8s-certs
+      readOnly: true
+    ... # omitted to save space
+```
+
+Similarly `kube-controller-manager`'s manifest would like:-
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  creationTimestamp: null
+  labels:
+    component: kube-controller-manager
+    tier: control-plane
+  name: kube-controller-manager
+  namespace: kube-system
+spec:
+  securityContext:
+    supplementalGroups:
+    - 2100
+  containers:
+  - name: kube-controller-manager
+    command:
+    - kube-controller-manager
+    - --cluster-signing-cert-file=/etc/kubernetes/pki/ca.crt
+    ...
+    # omitted to save space.
+    securityContext:
+      runAsUser: 2001  # this value is only an example and is not the id we plan to use.
+      runAsGroup: 2001 # this value is only an example and is not the id we plan to use.
+      allowPrivilegeEscalation: false
+      seccompProfile:
+        type: runtime/default
+      capabilities:
+        drop:
+        - ALL
+    image: k8s.gcr.io/kube-controller-manager:v1.21.0-beta.0.368_9850bf06b571d5-dirty
+    ... # omitted to save space.
+    volumeMounts:
+    - mountPath: /etc/kubernetes/pki
+      name: k8s-certs
+      readOnly: true
+    ... # omitted to save space. 
+```
+
+Here is a table of all the things that `kube-apiserver`, `kube-controller-manager`, `kube-scheduler` and `etcd` mount. 
+
+World readable stuff:
+- /usr/local/share/ca-certificates
+- /usr/share/ca-certificates
+- /etc/ssl/certs
+- /etc/ca-certificates
+
+<br />
+
+**Table key:-**
+
+| Abbreviation | Explanation |
+|--------------|-------------|
+| etcd | The UID/GID that we will assign to `etcd` |
+| kas  | The UID/GID that we will assign to `kube-apiserver` |
+| kcm | The UID/GID that we will assign to `kube-controller-manager` |
+| ks | The UID/GID that we will assign to `kube-scheduler` |
+| etcd_ca_crt_readers | The GID we will assign to a group that allows you to read /etc/kubernetes/pki/etcd/ca.crt |
+| ca_crt_readers | The GID we will assign to a group that allows you to read /etc/kubernetes/pki/etcd/ca.crt |
+| sa_key_readers | The GID we will assign to a group that allows you to read /etc/kubernetes/pki/sa.key |
+| front_proxy_ca_crt_readers | The GID we will assign to a group that allows you to read /etc/kubernetes/pki/front-proxy-ca.crt |
+
+<br/>
+
+**Stuff that we care about for this kep:-**
+| file/directory                                   | Components | File permission |
+| -------------------------------------------------|------------|-----------------|
```

Comment:
Is there a reason to limit the file permissions to owner read/write and group readonly i.e. 640. Would it break anything if we further limit it to 600 ?